pallets/staking/src/lib.rs

```diff
@@ -1734,6 +1734,54 @@ decl_module! {
 impl<T: Trait> Module<T> {
     // PUBLIC IMMUTABLES
 
+    /// POLYMESH-NOTE: This change is polymesh specific to query the list of all invalidate nominators
+    /// It is recommended to not call this function onchain. It is a non-deterministic function that is
+    /// suitable for offchain workers only.
+    pub fn fetch_invalid_cdd_nominators(buffer: u64) -> Vec<T::AccountId> {
+        let invalid_nominators = <Nominators<T>>::enumerate()
+            .into_iter()
+            .filter_map(|(nominator_stash_key, _nominations)| {
+                if let Ok(key_id) = AccountKey::try_from(nominator_stash_key.encode()) {
+                    if let Some(nominate_identity) = <identity::Module<T>>::get_identity(&(key_id))
+                    {
+                        if !(<identity::Module<T>>::fetch_cdd(
+                            nominate_identity,
+                            buffer.saturated_into::<T::Moment>(),
+                        ))
+                        .is_some()
+                        {
+                            return Some(nominator_stash_key);
+                        }
+                    }
+                }
+                return None;
+            })
+            .collect::<Vec<T::AccountId>>();
+        return invalid_nominators;
+    }
+
+    /// POLYMESH-NOTE: This is Polymesh specific change
+    /// Here we are assuming that passed targets are always be a those nominators whose cdd
+    /// claim get expired or going to expire after the `buffer_time`.
+    pub fn unsafe_validate_cdd_expiry_nominators(targets: Vec<T::AccountId>) -> DispatchResult {
```

Comment:
This gives validators the power to unilaterally remove all nominators (and validators) from the pool. This function should make sure that only nominators with expired cdd are being removed.